scoring/scoring.go

```diff
@@ -183,10 +183,16 @@ func MostGuessableMatchSequence(password string, matches []*match.Match, exclude
 			}
 
 			for k >= 0 {
-				m := optimal.m[k][l]
+				m, ok := optimal.m[k][l]
+				l--
+				if !ok {
+					//we're counting down through the potential keys (l). It's possible that
+					//the keys are non-contiguous so we need to skip values of l which aren't
+					//valid keys
+					continue
+				}
```

Comment:
From my previous understanding, this function should always receive contiguous matches since the missing space should be filled with bruteforce matches, so the real bug here might be that we have wrong indexes generated for some of the matches...
As far as I can tell this change only works around this bug but does not explain/fix the real issue (and it also  introduces the possibility of an infinite loop or a crash if `l` ever becomes negative ?)

Reply:
I can't see how we'd end up in an infinite loop in any which was not already present. I've only moved "l--" to before the hash lookup (so that we don't end up stuck at the same l value) which shouldn't change anything.

Reply:
Sorry you're right of course. I'll merge your change since it should never make the situation worse.

However I'd still like to understand how we get non-contiguous matches here, it looks like another bug...